### PR TITLE
do not set h2 alpn when downstream config is set

### PR DESCRIPTION
**Pull request by ramaraochavali**
This PR https://github.com/istio/istio/pull/31273 has incorrectly started setting h2 ALPN  when Downstream Config is set. This PR fixes that regression

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
lgtm,  can you add a test in case someone misset it again?

**Comment by ramaraochavali** [comment]
Sure. Will add

**Comment by ramaraochavali** [comment]
@hzxuzhonghu added tests and also fixed how newH2TestCluster is built so that `TestBuildUpstreamClusterTLSContext` validates its correctly.

File: pilot/pkg/networking/core/v1alpha3/cluster_builder_test.go [hunk 1]
```diff
@@ -2377,3 +2388,39 @@ func TestShouldH2Upgrade(t *testing.T) {
...
+		},
+	}
+
+	cb := NewClusterBuilder(nil, nil)
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			isHttp2Cluster := cb.IsHttp2Cluster(test.cluster)
+			if isHttp2Cluster != test.isHttp2Cluster {
+				t.Fatalf("got: %t, want: %t", isHttp2Cluster, test.isHttp2Cluster)
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster_builder_test.go:2422]: t.Errorf

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/cluster_builder_test.go:2422]: done

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**